Src/SpriteBatch.cpp

```diff
@@ -1036,10 +1054,29 @@ void XM_CALLCONV SpriteBatch::Begin(
     SpriteSortMode sortMode,
     FXMMATRIX transformMatrix)
 {
-    pImpl->Begin(
-        commandList,
-        sortMode,
-        transformMatrix);
+    pImpl->Begin(commandList, sortMode, transformMatrix);
+}
+
+
+_Use_decl_annotations_
+void XM_CALLCONV SpriteBatch::Begin(
+    ID3D12GraphicsCommandList* commandList,
+    D3D12_GPU_DESCRIPTOR_HANDLE sampler,
+    SpriteSortMode sortMode,
+    FXMMATRIX transformMatrix)
+{
+    if (!sampler.ptr)
+        throw std::exception("Invalid heap-based sampler for Begin");
+
+    if (!pImpl->mSampler.ptr)
+    {
+        DebugTrace("ERROR: sampler version of Begin requires SpriteBatch was created with a heap-based sampler\n");
+        throw std::exception("SpriteBatch::Begin");
+    }
+
+    pImpl->mSampler = sampler;
```

Comment:
I basically switch it to use the sampler you provided until the next time you provide a new one. This seemed easier than having some kind of 'stack' or revert to the original one.